src/cargo/core/compiler/unit_dependencies.rs

```diff
@@ -333,10 +368,12 @@ fn compute_deps<'a, 'cfg>(
                 .filter(|t| {
                     let no_required_features = Vec::new();
 
+                    let features = resolve_all_features(state.resolve(), state.features(), id);
```

Comment:
Do you think this function name is suffcient to convey the intent of the code? @alexcrichton, if you are still surprised, can you tell me in what perspective?

Reply:
I suspect the function will get rewritten (and renamed) as a result of #7950, so I wouldn't worry about it for now.

For performance reasons, this should be rearranged so that it only calls `resolve_all_features` iff it needs to (a binary with required-features).

Reply:
I'll test whether `require_feature` is empty, and if target is bin, should I only test for bins? Are there any other target (e.g. examples, tests) I should take into consideration?

Reply:
Or should I just check `required_features`?

Reply:
Oh, I see.